Add overview docs (#553)
MicrosoftDocs/quantum-docs-pr: articles/overview/what-is-quantum-computing.md

---
title: What is quantum computing?
description: 
author: natke
ms.author:  
ms.date: 10/14/2019
ms.topic: article
uid: microsoft.quantum.overview.whatis
---

# What is quantum computing?

Quantum computing is the application of quantum physics to the field of computer science. Quantum computing might sound daunting, but with the right resources you can start building quantum applications today.

## The qubit

> The basic unit of quantum computing is the `qubit`. Unlike a `bit` in classical computing, which can be `0` or `1`, a qubit can be in an infinite number of states - a **superposition** of both 0 and 1.

Reviewer: Quick question, but do we want to use code formatting for "qubit" and "bit"? With respect to the technical content, while it is definitely true and correct that a qubit can be in an infinite number of states, I've generally found that fact can be fairly easily misunderstood such that I would suggest either more qualifications to the statement here or deferring that to the quantum concepts guide.

Author: Good question, we should decide on a standard here. The fixed width formatting is used for more than code, often used to denote terminology (and if a repo is localized then these terms do not get translated)

The act of **measurement** changes a qubit. It goes from being in the infinite states of superposition to one of the binary states of 0 or 1. Again, measurement of a qubit is unlike a classical bit, which when 0 is measured as 0, and when 1 is measured as 1.

Qubits can also be **entangled**: when we make a measurement of one entangled qubit, it affects the state of the other, even if the qubits are physically far apart.

> 

Reviewer: At this point, superposition, measurement, and entanglement probably feel a bit mysterious, such that it might be helpful emphasizing that we can understand these quantum effects through working with the Quantum Development Kit or through looking at the math, whichever we prefer.

Author: Yes, I agree! I would love to come up with a terse example but have so far failed.

Reviewer: Let me think a bit on that...

Reviewer 2: I agree, interference and entanglement are concepts difficult to grasp. I don't know if we should adress them in the introduction, they can be confusing and mysterious. But on the other hand they are very important to understand the reason of quantum computing.

## What can a quantum computer do?

Quantum algorithms can be designed to take advantage of qubit superposition and entanglement to speed up classical algorithms, or to provide entirely new ways of modeling physical systems.

> One of the most famous quantum algorithms is `Shor's algorithm` for factorization, which makes the classically intractable problem of factorization of a large number into two prime numbers fast enough to challenge traditional cryptography.

Reviewer: Similarly here, do we want "Shor's algorithm" to be typeset as a code literal? It may also be helpful to link to the relevant sample for integer factorization.

Reviewer 2: I don't know if we should use this space to explain what is already explained in the section "What can we do with quantum computers".  Maybe we should save this space to explain "why quantum computers work" or a bit of the circuit model.

`Grover's algorithm` provides a quadratic factor speed-up for database searching.

Using quantum programs to model quantum systems themselves is another application of quantum computing. Photosynthesis, superconductors, and chemical molecules are examples of quantum systems that can be simulated using quantum programs.

## Quantum hardware

In classical computers, bits correspond to voltage levels in silicon circuits. Quantum computing hardware can be implemented by many different physical realizations of qubits: trapped ions, superconducting, neutral atoms, electron spin, light polarization, topological qubits. Quantum hardware is an emergent technology. It requires balancing fidelity of the system with scalability. The larger the scale (that is, number of qubits), the higher the error rate.

## Quantum simulations

Up to a certain scale (number of qubits), quantum computing can be simulated on a classical computer. Using simulations is how you can start to write quantum programs using Microsoft Quantum, the Quantum Development Kit (QDK), and the Q# language.

## Get started today

* Build a quantum program that implements [Grover's algorithm](quickstarts/grovers.md)

> * Take a quick look at Q# by developing a [Bell State](quickstart.md) program

Reviewer: Should these be xref links rather than explicit paths?

Author: Yes, they should!

---
title: What is quantum computing?
description: Learn what quantum computing is, and what a quantum computer can do
author: natke
ms.author: nakersha
ms.date: 10/14/2019
ms.topic: article
uid: microsoft.quantum.overview.what
---

# What is quantum computing?

> Quantum computing is the application of quantum physics to the field of computer science.

Reviewer 2: ```suggestion
Quantum computing is a model of computation based on the underlying logic of quantum physics.
```

> The theory of quantum physics posits that matter, at a quantum level is simultaneously in multiple states. And those many states interfere with each other like waves in a tide pool.

Reviewer 2: ```suggestion
The theory of quantum physics posits that matter, at a quantum level can be in a superposition of multiple classical states. And those many states interfere with each other like waves in a tide pool.
```
We have to be careful with the term "state". Saying that matter can be in "multiple states" can lead to misconceptions because we physicists usually refer to "state of the system" as the quantum superposition of multiple classical states. Therefore, the system is only in one state (that is a specific superposition of several classical states).

Quantum computing stores information in quantum states of matter and uses quantum operations to compute on that information, by harnessing and learning to program quantum interference.

> Quantum computing might sound daunting, but with the right resources you can start building quantum applications today.

Reviewer 3: Don't know why, but I am inspired and ready to go. Honestly, this sentence is 🔥 .

## The qubit

> In quantum computing, a quantum bit is a unit of quantum information, like a classical bit. Where classical bits hold a single binary value such as a 0 or 1, a qubit can hold both values at the same time. This is called **superposition**.

Reviewer 3: ```suggestion
In quantum computing, a quantum bit (hence, "qubit") is a unit of quantum information, like a classical bit. Where classical bits hold a single binary value such as a 0 or 1, a qubit can hold both values at the same time. This is called **superposition**.
```

Author: Fixed

> The act of **measurement** changes a qubit. It goes from being in the infinite states of superposition to one of the binary states of 0 or 1. Again, measurement of a qubit is unlike a classical bit, which when 0 is measured as 0, and when 1 is measured as 1.

Reviewer 2: ```suggestion
The act of **measurement** changes a qubit. It goes from being in a superposition of different states to one of the binary states of 0 or 1. Again, measurement of a qubit is unlike a classical bit, which when 0 is measured as 0, and when 1 is measured as 1.
```
The qubit is not in infinite states of superposition but just in one superposition. And if the qubit is good, which means that only has two possible states, then the superposition is only of two states and not infinitely many.

> The act of **measurement** changes a qubit. It goes from being in the infinite states of superposition to one of the binary states of 0 or 1. Again, measurement of a qubit is unlike a classical bit, which when 0 is measured as 0, and when 1 is measured as 1.

Reviewer 3: I'll leave the actual phrasing up to y'all, but figured I'd try to clarify what Eduardo's getting at for you @natke . Basically, in a single qubit, the superposition will always be just between |0> and |1>. _However_, the superposition can be arbitrarily weighted between the two (think of a unit vector in the xy plane pointing in any direction between -1 and 1) so that's where the "infinitely many" possible states comes from. 

(Just to cover my keister in case Eduardo/Chris see this, it's of course not just the xy plane, but the full 3 dimensions; hence the Bloch sphere representation)

Qubits can also be **entangled**: when we make a measurement of one entangled qubit, it affects the state of the other, even if the qubits are physically far apart.

## What can a quantum computer do?

When multiple qubits act coherently, they can process multiple options simultaneously. This allows them to process information in a fraction of the time it would take even the fastest nonquantum systems.

Quantum algorithms can be designed to take advantage of qubit superposition and entanglement to speed up classical algorithms, or to provide entirely new ways of modeling physical systems.

One of the most famous quantum algorithms is `Shor's algorithm` for factorization, which makes the classically intractable problem of factorization of a large number into two prime numbers fast enough to challenge traditional cryptography.

On the more constructive side, algorithms for secure cryptographic key distribution are made possible by superposition, quantum entanglement, and the **no cloning** property of qubits: the inability for qubits to be copied without detection.

> `Grover's algorithm` provides a quadratic factor speed-up for database searching.

Reviewer: Similarly here, looks like code fences are being used when we likely want emphasis?

> Using quantum programs to model quantum systems themselves is another application of quantum computing. Photosynthesis, superconductors, and chemical molecules are examples of quantum systems that can be simulated using quantum programs.

Reviewer 3: "chemical molecules" feels redundant---maybe "complex molecules"?

Author: Fixed

## Quantum hardware

> In classical computers, bits correspond to voltage levels in silicon circuits. Quantum computing hardware can be implemented by many different physical realizations of qubits: trapped ions, superconducting, neutral atoms, electron spin, light polarization, topological qubits. Quantum hardware is an emergent technology. It requires balancing fidelity of the system with scalability. The larger the scale (that is, number of qubits), the higher the error rate.

Reviewer 2: The beginning of this paragraph suggest that all classic computers are done with transistors, which maybe can lead to confusion. You can construct classical computers with any system that behaves according classical mechanics, for example with marbles or vaccum tubes. 
Personally I like the explanation how we use classical systems to represent classical bits and implement classical gates to then do the same for quantum computing but using quantum system to represent qubits and quantum gates.

Reviewer 3: I like the conciseness/focus of the paragraph, so I'd personally say don't expand beyond ~one sentence referring to the classical stuff. But to clear up potential confusion it might help to just say something like the 1's and 0's of bits correspond to the on/off's of voltage levels (thanks to transistors). Because people might scratch their head and think "but wait, voltage levels can be in many different states too" and then oh no we'd have to explain analog computing too.

## Quantum simulations

Up to a certain scale (number of qubits), quantum computing can be simulated on a classical computer. Using simulations is how you can start to write quantum programs using Microsoft Quantum, the Quantum Development Kit (QDK), and the Q# language.

## Get started today

* Build a quantum program that implements [Grover's algorithm](xref:microsoft.quantum.quickstarts.search)

> * Take a quick look at Q# by developing a [Bell State](xref:microsoft.quantum.write-program) program

Reviewer 2: Same as in the other sections, do we want external links inside the guide? Maybe avoiding them until the end helps people to focus only on the guide.

> * Take a quick look at Q# by developing a [Bell State](xref:microsoft.quantum.write-program) program

Reviewer 3: Perhaps toss the hyperlinks on the whole sentences or just the first phrase? With it on just that the key word it sort of seems like it'll be a hyperlink to the definition rather than a quickstart
